File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
    int num_rows = matrix.length;
    int num_cols = matrix[0].length;
    Stack<Integer> stack1=new Stack<>();
    Stack<Integer> stack2=new Stack<>();
    for(int e=0;e<num_rows;e++){
        for(int f=0;f<num_cols;f++){
            if(matrix[e][f]==0){
                stack1.push(e);
                stack2.push(f);
            }
        }
    }
    if(stack1.size()==(num_cols*num_rows)) return;
    while(!(stack1.empty())){
        int e=stack1.pop();
        int f=stack2.pop();
        for(int i=e;i-1>-1;i--){
            matrix[i-1][f]=0;
        }
        for(int i=f;i+1<num_cols;i++){
            matrix[e][i+1]=0;
        }
        for(int i=e;i+1<num_rows;i++){
            matrix[i+1][f]=0;
        }
        for(int i=f;i-1>-1;i--){
            matrix[e][i-1]=0;
        }
    }    
    }
}

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cpp
class Solution {
private:
    void setKaro(vector<vector<int>>& matrix,int e,int f){
        int n=matrix.size();
        int m=matrix[0].size();
        for(int i=0;i<n;i++){
            if(matrix[i][f]!=0) matrix[i][f]=0;
        }
        for(int i=0;i<m;i++){
            if(matrix[e][i]!=0) matrix[e][i]=0;
        }
    }
public:
    void setZeroes(vector<vector<int>>& matrix) {
        vector<pair<int,int>> x;
        for(int e=0;e<matrix.size();e++){
            for(int f=0;f<matrix[0].size();f++){
                if(matrix[e][f]==0) x.push_back(make_pair(e,f));
            }
        }
        for(int e=0;e<x.size();e++){
            setKaro(matrix,(x[e].first),(x[e].second));
        }
    }
};